components/brave_rewards/resources/ui/reducers/rewards_reducer.ts

```diff
@@ -129,6 +129,121 @@ const rewardsReducer: Reducer<Rewards.State | undefined> = (state: Rewards.State
       state.adsData.adsIsSupported = action.payload.adsData.adsIsSupported
       break
     }
+    case types.GET_ADS_HISTORY: {
+      chrome.send('brave_rewards.getAdsHistory')
+      break
+    }
+    case types.ON_ADS_HISTORY: {
+      if (!action.payload.adsHistory) {
+        break
+      }
+
+      state = { ...state }
+
+      if (!state.adsHistory) {
+        state.adsHistory = defaultState.adsHistory
+      }
+
+      state.adsHistory = action.payload.adsHistory
+      break
+    }
+    case types.TOGGLE_AD_THUMB_UP: {
+      chrome.send('brave_rewards.toggleAdThumbUp',
+                  [action.payload.uuid, action.payload.creativeSetId, action.payload.likeAction])
+      break
+    }
+    case types.ON_TOGGLE_AD_THUMB_UP: {
+      state = { ...state }
+      state.adsHistory.forEach((history: Rewards.AdsHistoryData) => {
```

Comment:
my suggestion would be that on all this actions where we change some thing just call `getAdsHistory`. This way we can't be out of sync

Reply:
Yeah, that's a really good idea. Will fix.

Reply:
Fixed.